File: app/src/main/java/com/example/turnovermanagment/Data/QuestionsQueueService.java
package com.example.turnovermanagment.Data;

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.catch
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class QuestionsQueueService(private val databaseManager:DatabaseManager) {

    fun submitQuestion(question: Question): Flow<Result<Unit>> = flow {
        emit(Result.success(databaseManager.addQuestion(question).await()))
    }.catch { e ->
            emit(Result.failure(e))
    }

    fun retrieveQuestions(): Flow<Result<List<Question>>> = flow {
        emit(Result.success(databaseManager.getQuestions().await()))
    }.catch { e ->
            emit(Result.failure(e))
    }

    fun respondToQuestion(questionId: String, response: String): Flow<Result<Unit>> = flow {
        emit(Result.success(databaseManager.updateQuestion(questionId, response).await()))
    }.catch { e ->
            emit(Result.failure(e))
    }

    fun prioritizeQuestion(questionId: String, priority: Int): Flow<Result<Unit>> = flow {
        emit(Result.success(databaseManager.updateQuestionPriority(questionId, priority).await()))
    }.catch { e ->
            emit(Result.failure(e))
    }

    fun archiveQuestion(questionId: String): Flow<Result<Unit>> = flow {
        emit(Result.success(databaseManager.archiveQuestion(questionId).await()))
    }.catch { e ->
            emit(Result.failure(e))
    }
}

File: app/src/main/java/com/example/turnovermanagment/Data/QuestionsQueueService.kt
package com.example.turnovermanagment.Data


import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn




class QuestionQueueService(private val databaseManager: DatabaseManager) {

    fun addQuestion(question: Question) = flow {
        emit(databaseManager.addQuestion(question))
    }.flowOn(Dispatchers.IO)

    fun retrieveQuestions() = flow {
        emit(databaseManager.getQuestions())
    }.flowOn(Dispatchers.IO)

    fun updateQuestion(questionId: String, updates: Map<String, Any>) = flow {
        emit(databaseManager.updateQuestion(questionId, updates))
    }.flowOn(Dispatchers.IO)

    fun prioritizeQuestion(questionId: String, priority: Map<String, Int>) = flow {
        val updates = mapOf("priority" to priority)
        emit(databaseManager.updateQuestion(questionId, updates))
    }.flowOn(Dispatchers.IO)

    fun archiveQuestion(questionId: String) = flow {
        emit(databaseManager.archiveQuestion(questionId))
    }.flowOn(Dispatchers.IO)
}












